libnest/src/config_parser.rs

```diff
@@ -0,0 +1,146 @@
+//! Nest configuration file parsing implementation.
+//!
+//! All the code fetching the settings from the config file is here, in
+//! `ConfigParser::load_to_config`.
+//! If settings are added, this is where code must be added to fetch these new settings.
+//!
+//!
+
+extern crate toml;
+
+use std::fs::File;
+use std::io::BufReader;
+use std::io::prelude::Read;
+use std::io;
+use std::path::PathBuf;
+use std::fmt;
+
+use config::Config;
+use repository::{Mirror, Repository};
+
+pub(crate) enum ParseConfError {
```

Comment:
We wan't this error to be accessible to the caller (see other comments), so this shouldn't be `pub(crate)`.

In a general way, try to avoid `pub(crate)`, use the module system of rust instead (`pub`) and keep them only for the extreme cases.